ldms/src/ldmsd/ldmsd_request.c

```diff
@@ -6008,6 +6010,183 @@ static int greeting_handler(ldmsd_req_ctxt_t reqc)
 	return 0;
 }
 
+static int dump_cfg_handler(ldmsd_req_ctxt_t reqc)
+{
+	FILE *fp;
+	char *filename = NULL;
+        extern struct plugin_list plugin_list;
+        struct ldmsd_plugin_cfg *p;
+        int rc;
+	int i;
+        reqc->errcode = 0;
+	filename = ldmsd_req_attr_str_value_get_by_id(reqc, LDMSD_ATTR_STRING);
+	fp = fopen(filename, "w+");
+
+	/* Auth */
+	ldmsd_auth_t auth;
+	ldmsd_cfg_lock(LDMSD_CFGOBJ_AUTH);
+	for (auth = (ldmsd_auth_t)ldmsd_cfgobj_first(LDMSD_CFGOBJ_AUTH); auth;
+			auth = (ldmsd_auth_t)ldmsd_cfgobj_next(&auth->obj)) {
+		fprintf(fp, "auth_add name=%s plugin=%s", auth->obj.name, auth->plugin);
+		if (auth->attrs) {
+			for (i = 0; i < auth->attrs->count; i++) {
+				struct attr_value *v = &auth->attrs->list[i];
+				fprintf(fp, " %s=%s", v->name, v->value);
+			}
+		}
+		fprintf(fp, "\n");
+	}
+	ldmsd_cfg_unlock(LDMSD_CFGOBJ_AUTH);
+	/* Listeners */
+	ldmsd_listen_t listen;
+	for (listen = (ldmsd_listen_t)ldmsd_cfgobj_first(LDMSD_CFGOBJ_LISTEN); listen;
+			listen = (ldmsd_listen_t)ldmsd_cfgobj_next(&listen->obj)) {
+		fprintf(fp, "listen xprt=%s port=%d",
+			listen->xprt,
+			listen->port_no);
+		if (listen->host)
+			fprintf(fp, " host=%s", listen->host);
+		if (listen->auth_name) {
+			if (listen->auth_dom_name)
+				fprintf(fp, " auth=%s", listen->auth_dom_name);
+			else
+				fprintf(fp, " auth=DEFAULT");
```

Comment:
The reason this is included is because this configuration assumes that no configuration options will be specified when starting a ldmsd with the "dumped" configuration files. I meant to make a note of this in the top of each configuration file, but dropped the ball. I'm open to a discussion about whether this is the correct approach or not.